### PR TITLE
Add SourceSelectionCriteria to replication config

### PR DESCRIPTION
**Pull request by poornas**
This PR is needed for viewing this additional config in mc

### CODE REVIEW

File: pkg/replication/replication.go [hunk 1]
```diff
@@ -195,6 +195,10 @@ func (c *Config) AddRule(opts Options) error {
 		},
 		DeleteMarkerReplication: DeleteMarkerReplication{Status: dmStatus},
 		DeleteReplication:       DeleteReplication{Status: vDeleteStatus},
+		SourceSelectionCriteria: SourceSelectionCriteria{
+			ReplicaModifications: ReplicaModifications{
+				Status: Enabled},
```
Comment (harshavardhana) [inline pkg/replication/replication.go:200]:
```suggestion
				Status: Enabled,
				},
```

Comment (harshavardhana) [reply pkg/replication/replication.go:200]: BTW should we do this by default?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: pkg/replication/replication.go (hunk 1 above)
Comment (poornas) [reply pkg/replication/replication.go]: If we do not enable this by default, we will need additional flags in mc to support toggling this status. It seems to make sense for bi-directional replication to keep source and replica in sync

Comment (harshavardhana) [reply pkg/replication/replication.go]: Hmm good - add a comment on why we enforce it @poornas - and also why it makes sense by referencing the link from AWS - such that our choice is to make it default

Comment (poornas) [reply pkg/replication/replication.go]: done

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**